todo_app/app.py

```diff
@@ -21,21 +37,60 @@ def create_app():
     app.config.from_object('todo_app.flask_config.Config')
     # All the routes and setup code etc
     @app.route('/')
+    @login_required
     def index(): 
         items = Get_items_from_mongodb()
         item_view_model = ViewModel(items)
-        return render_template("index.html", view_model=item_view_model)
+        return render_template("index.html", view_model=item_view_model,role=current_user.role)
         
     @app.route('/', methods=['POST'])
     def add_todo(): 
+        if current_user.role=='read_user':
+            return 'Forbidden',401
         item = request.form.get('additem')
         Create_iten_from_mongodb(item)
         return redirect('/')
 
     @app.route('/move', methods=['POST'])
     def Move_InTrello(): 
+        if current_user.role=='read_user':
+            return 'Forbidden',401
         item = request.form.get('id')
         move_item_from_mongodb(item)
         return redirect('/')
 
+    @app.route('/login/callback')
+    def CallbackLogin():
+        args = request.args
+        code= args.get("code")
+        username=''
+        url = 'https://github.com/login/oauth/access_token'
+        urluser='https://api.github.com/user'
+        myobj = {'client_id': 'b0c63e19e26853b4d122', 'client_secret':os.getenv('ClientSecret'),'code':code}
```

Comment:
Would be nice for client id to also be an environment variable. It's not exactly sensitive, but it is something we want to vary between environments. Currently deploying this application to different environments requires Python code changes, so it cannot work both locally and on Heroku or Azure.